Restore feature states requires restricted permissions (#3857)
elastic/docs-content: deploy-manage/tools/snapshot-and-restore/restore-snapshot.md

If you restore a snapshot’s cluster state, the operation restores all feature states in the snapshot by default. Similarly, if you don’t restore a snapshot’s cluster state, the operation doesn’t restore any feature states by default. You can also choose to restore only specific feature states from a snapshot, regardless of the cluster state.

> Feature backing indices are version dependent. To see which indices are included within a snapshot's feature state, [list an applicable snapshot](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-snapshot-get). For example on {{ech}} you might poll its latest snapshot on its built-in SLM policy `cloud-snapshot-policy`.

Reviewer: ```suggestion
Feature backing indices are version dependent. To see which indices are included within a snapshot's feature state, [list an applicable snapshot](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-snapshot-get). For example, on {{ech}} you might poll its latest snapshot on its built-in  `cloud-snapshot-policy` SLM policy:
```

```console
GET /_snapshot/_all/_all?filter_path=snapshots.feature_states&index_names=false&sort=start_time&size=1&order=desc&slm_policy_filter=cloud-snapshot-policy
```

> The response’s `feature_states` property contains a list of features in the snapshot as well as each feature’s indices. For an example cluster, its output might appear:

Reviewer: ```suggestion
The response’s `feature_states` property contains a list of features in the snapshot as well as each feature’s indices. The following is an example of the output that might display for a cluster:
```

            ".kibana_usage_counters_9.x.x_001",
            ".kibana_9.x.x_001",
            ".apm-custom-link",
            ".kibana_search_solution_9.x.x_001",
            ".kibana_task_manager_9.x.x_001",
            ".apm-agent-configuration",
            ".kibana_locks-000001",
            ".kibana_security_session_1",
            ".kibana_alerting_cases_9.x.x_001",
            ".kibana_analytics_9.x.x_001",
            ".kibana_security_solution_9.x.x_001",
            ".kibana_ingest_9.x.x_001"
          ]
        }
      ]
    }
  ]
}
```

To restore a specific feature state from the snapshot, specify the `feature_name` from the response in the [restore snapshot API’s](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-snapshot-restore) `feature_states` parameter.

Feature state names may line-up to {{kib}} UI sections while not being encompassing to completely reset that UI. For example, the {{fleet}} UI backs against the `fleet` feature state as well as `kibana` and `security`. It is important to ensure you line up desired reset behavior with which backing feature states to include in the `feature_states` parameter when restoring.

::::{warning}

> Restoring the `security` feature state overwrites system indices used for authentication. If you use {{ech}} or {{ece}}, ensure you have access to the [{{es}} API console](cloud://reference/cloud/cloud-hosted/ec-api-console.md) before restoring the `security` feature state. If you run {{es}} on your own hardware or in {{eck}}, [create a temporary user with elevated permissions to edit restricted indices in the file realm](/troubleshoot/elasticsearch/file-based-recovery.md) to ensure you’ll still be able to access your cluster.

Reviewer: I think the path includes an extra repo when linking to this page: https://github.com/elastic/cloud/blob/master/docs/reference/cloud-hosted/ec-api-console.md?plain=1
